File: SpendingTracker.Droid/MainActivity.cs
using Android.App;
using Android.Widget;
using Android.OS;
using Xamarin;
using SpendingTracker;
using System.IO;
using Xamarin.Forms.Platform.Android;
using SpendingTracker;
using System;
using Microsoft.FSharp.Core;


namespace SpendingTracker.Droid
{
	public static class ToFSharpFuncConverterExtensions
	{
		private static readonly Unit Unit = (Unit)Activator.CreateInstance(typeof(Unit), true);

		public static Func<T, Unit> ToFunc<T>(this Action<T> action)
		{
			return x => { action(x); return Unit; };
		}

		public static FSharpFunc<T, Unit> ToFSharpFunc<T>(this Action<T> action)
		{
			return FSharpFunc<T, Unit>.FromConverter(new Converter<T, Unit>(action.ToFunc()));
		}
	}

	[Activity(Label = "SpendingTracker.Droid", Icon = "@mipmap/icon", MainLauncher = true)]
	public class MainActivity : FormsAppCompatActivity
	{
		protected override void OnCreate(Bundle savedInstanceState)
		{
			ToolbarResource = Resource.Layout.toolbar;
			TabLayoutResource = Resource.Layout.tabs;

			base.OnCreate(savedInstanceState);
			Xamarin.Forms.Forms.Init(this, savedInstanceState);

			var path = System.Environment.GetFolderPath(System.Environment.SpecialFolder.Personal);
			Action<string> exporter = null;
			//= dbPath =>
			//{
			//using (var output = File.Create())
			//{
			//	using (FileStream fs = File.OpenRead(dbPath))
			//	{
			//		while (fs.CanRead)
			//		{
			//			output.WriteByte((byte)fs.ReadByte());
			//		}
			//		fs.Close();
			//	}
			//	output.Close();
			//}				
			//};
			//var result = exporter.ToFSharpFunc();


			//LoadApplication(new FormsApp(path, new SQLitePCL.SQLite3Provider_e_sqlite3(), null));
		}

	}
}

File: Droid/MainActivity.fs
namespace SpendingTracker.Droid
open System;
open System.IO
open Android.App;
open Android.Content;
open Android.Content.PM;
open Android.Runtime;
open Android.Views;
open Android.Widget;
open Android.OS;
open Xamarin
open Xamarin.Forms.Platform.Android
open BarChart

[<Activity (Label = "SpendingTracker.Droid", Icon = "@mipmap/icon", MainLauncher = true, Theme = "@style/Theme.Splash")>]
type MainActivity() =
    inherit FormsAppCompatActivity()
    override this.OnCreate (bundle: Bundle) =
      this.SetTheme(Resource_Style.MyTheme)
      base.OnCreate (bundle)


      //FormsAppCompatActivity.ToolbarResource <- Resource_Layout.toolbar
      //FormsAppCompatActivity.TabLayoutResource <- Resource_Layout.tabs


      let data = [| 1.f; 2.f; 4.f; 8.f; 16.f; 32.f; |] |> Array.map (fun x -> new BarModel(Value = x))
      let chart = new BarChartView (this, ItemsSource = data);

      this.AddContentView (chart, new ViewGroup.LayoutParams (ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.MatchParent));


        //Xamarin.Forms.Forms.Init (this, bundle)

        //let path = System.Environment.GetFolderPath(System.Environment.SpecialFolder.Personal)
        //let write p data = File.WriteAllBytes(p, data)
        //let exportDb name = 
        //  let writePath = Path.Combine(Android.OS.Environment.ExternalStorageDirectory.Path, "database.db")
        //  File.ReadAllBytes(Path.Combine(path, name)) |> write writePath
        //  Toast.MakeText(this, "Db exported to storage", ToastLength.Short).Show()

        //let createApp p dbType exporter = new SpendingTracker.FormsApp(p, dbType, exporter)


        //let app =  createApp path (new SQLitePCL.SQLite3Provider_e_sqlite3()) exportDb
        //this.LoadApplication(app)